teamcity: failed test: version-upgrade

The following tests appear to have failed on master (roachtest): acceptance/version-upgrade

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+version-upgrade).

[#1018742](https://teamcity.cockroachdb.com/viewLog.html?buildId=1018742):


```
acceptance/version-upgrade
--- FAIL: roachtest/acceptance/version-upgrade (76.323s)
	test.go:645,upgrade.go:393,upgrade.go:395,upgrade.go:546,acceptance.go:65: dial tcp 127.0.0.1:26259: connect: connection refused





```

Please assign, take a look and update the issue accordingly.

`n2` [crashed](https://teamcity.cockroachdb.com/repository/download/Cockroach_UnitTests_Roachtest/1018737:id/acceptance/version-upgrade/2.logs/cockroach-v200.aa65caef342d.roach.2018-11-19T18_26_55Z.024594.log): 

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12ee897]

goroutine 1129 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc4204e6090, 0x26120c0, 0xc420873980)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:177 +0x11f
panic(0x209b900, 0x3465aa0)
	/usr/local/go/src/runtime/panic.go:505 +0x229
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*Progress).maybeUpdate(...)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/progress.go:120
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).appendEntry(0xc42579a500, 0xc426546f00, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:608 +0xe7
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).becomeLeader(0xc42579a500)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:710 +0x258
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).campaign(0xc42579a500, 0x232ec3a, 0x10)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:733 +0x62f
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).campaign(0xc42579a500, 0x233440b, 0x13)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:731 +0x690
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).Step(0xc42579a500, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:851 +0x2ff
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*RawNode).Campaign(0xc4256434c0, 0xc420c22570, 0x3)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/rawnode.go:140 +0x61
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroupLocked(0xc420c1ca80, 0x2418801, 0x24182f8, 0x3a38678, 0x3a38678)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:598 +0x3a2
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).maybeInitializeRaftGroup(0xc420c1ca80, 0x26120c0, 0xc420873980)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1843 +0x194
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Send(0xc420c1ca80, 0x26120c0, 0xc420873980, 0x156899da34a5061e, 0x0, 0x0, 0x0, 0x9f, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1867 +0x136
github.com/cockroachdb/cockroach/pkg/storage.(*pendingLeaseRequest).requestLeaseAsync.func2(0x26120c0, 0xc420873980)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_range_lease.go:310 +0x5af
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc4204e6090, 0x26120c0, 0xc420873980, 0xc4268528c0, 0x35, 0x262fec0, 0xc4202e8540, 0xc424de7710)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:263 +0xe6
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:258 +0x133
```

The node was running `v2.0.0`. This is ringing a bell for me. I think we might have seen this crash before.

--- a/pkg/cmd/roachtest/upgrade.go
+++ b/pkg/cmd/roachtest/upgrade.go
@@ -280,7 +280,7 @@ func runVersionUpgrade(ctx context.Context, t *test, c *cluster) {
 
 	// versionStep is an isolated version migration on a running cluster.
 	type versionStep struct {
-		clusterVersion string
+		clusterVersion string // if empty, use crdb_internal.node_executable_version
 		run            func()
 	}
 
@@ -364,6 +364,16 @@ func runVersionUpgrade(ctx context.Context, t *test, c *cluster) {
 		return versionStep{
 			clusterVersion: newVersion,
 			run: func() {
+				func() {
+					if newVersion != "" {
+						return
+					}
+					db1 := c.Conn(ctx, 1)
+					defer db1.Close()
+					if err := db1.QueryRow(`SELECT crdb_internal.node_executable_version()`).Scan(&newVersion); err != nil {
+						t.Fatal(err)
+					}
+				}()
 				t.l.Printf("%s: cluster\n", newVersion)
 
 				// hasShowSettingBug is true when we're working around
@@ -437,20 +447,17 @@ func runVersionUpgrade(ctx context.Context, t *test, c *cluster) {
 
 	const baseVersion = "v1.0.6"
 	steps := []versionStep{
-		// v1.1.0 is the first binary version that knows about cluster versions,
-		// but thinks it can only support up to 1.0-3.
-		binaryVersionUpgrade("v1.1.0", nodes),
-		clusterVersionUpgrade("1.0", true /* manual */),
-		clusterVersionUpgrade("1.0-3", true /* manual */),
-
-		binaryVersionUpgrade("v1.1.1", nodes),
+		// v1.1.0 is the first binary version that knows about cluster versions.
+		binaryVersionUpgrade("v1.1.9", nodes),
 		clusterVersionUpgrade("1.1", true /* manual */),
 
-		binaryVersionUpgrade("v2.0.0", nodes),
-		clusterVersionUpgrade("1.1-6", true /* manual */),
+		// NB: v2.0.6 doesn't have https://github.com/cockroachdb/cockroach/issues/31380,
+		// so this acceptance test will fail on OSX Mojave. v2.0.7 will have the patch.
+		binaryVersionUpgrade("v2.0.6", nodes),
 		clusterVersionUpgrade("2.0", true /* manual */),
 
 		binaryVersionUpgrade("HEAD", nodes),
+		clusterVersionUpgrade("", false /* manual */),
 	}
 
 	type feature struct {

